File: Control/MainController.cpp
#include "MainController.h"
#include <QImage>
#include <opencv2/opencv.hpp>
using namespace cv;
using namespace std;

MainController::MainController(QObject *parent) : QObject(parent),
    mMainLoopTimer(this)
{
    MyDebug;
    initConnect();
}

///
/// \brief 控制器启动入口
///
void MainController::start()
{
    MyDebug;
    mViewController.start();
    mMainLoopTimer.start(50);

    mViewController.navigateTo("TestCameraWin");
    mCameraController.startCamera(0, X_USB_CAM);
}



void MainController::initConnect()
{
    MyDebug;
    connect(&mMainLoopTimer, SIGNAL(timeout()), this, SLOT(mainLoop()));
}

///
/// \brief 主循环信号槽
///
void MainController::mainLoop()
{
    MyDebug;
    Mat cam_frame;
    mCameraController.gotFrame(0, cam_frame);
    mViewController.displayImageInQLabel(cam_frame, mViewController.mTestMovieShow);
    mViewController.mTestMovieShow->update();
}

File: Control/MainController.py
"""
       .==.        .==.
      //`^\\      //^`\\
     // ^ ^\(\__/)/^ ^^\\
    //^ ^^ ^/+  0\ ^^ ^ \\
   //^ ^^ ^/( >< )\^ ^ ^ \\
  // ^^ ^/\| v''v |/\^ ^ ^\\
 // ^^/\/ /  `~~`  \ \/\^ ^\\
 ----------------------------
BE CAREFULL! THERE IS A DRAGON.

Function：The controller of the app

Modules：
pass

(c) 肖镇龙(xzl) 2021

Dependencies：

Updating Records:
2021-01-22 09:38:15 xzl
"""
from PyQt5.QtWidgets import QApplication, QWidget
from PyQt5 import QtCore, QtWidgets, QtGui, uic
from Views import MainWindow
from Common.DebugPrint import myDebug, get_current_function_name
from Control import CameraController, ViewController
import sys
sys.path.append("../")

class MainController(QtCore.QObject):
    def __init__(self, *arg):
        myDebug(self.__class__.__name__, get_current_function_name())
        super(MainController, self).__init__(*arg)
        self.mMainLoopTimer = QtCore.QTimer(self)
        self.mViewController = ViewController.ViewController()
        self.initConnect()

    def start(self):
        myDebug(self.__class__.__name__, get_current_function_name())
        self.mViewController.start()
        self.mMainLoopTimer.start(50)

        self.mViewController.navigateTo("ContentsNavWin")

    def initConnect(self):
        myDebug(self.__class__.__name__, get_current_function_name())
        self.mMainLoopTimer.timeout.connect(self.mainLoop)

    def mainLoop(self):
        # myDebug(self.__class__.__name__, get_current_function_name())
        pass
    